### ISSUE TITLE
Cocoapods errors when building to iOS after installing any package via Package Manager

### ISSUE THREAD

**Issue by jonahgoldsaito** [firebase/quickstart-unity#700]
### Please fill in the following fields:
Unity editor version: 19.3.8
Firebase Unity SDK version: 6.15.0 (and 6.14.1)
Source you installed the SDK (.unitypackage or Unity Package Manager): Unity Package Manager
Firebase plugins in use (Auth, Database, etc.): Firebase Auth (I've stripped out others for debugging)
Additional SDKs you are using (Facebook, AdMob, etc.): None
Platform you are using the Unity editor on (Mac, Windows, or Linux): Mac OSX Catalina
Platform you are targeting (iOS, Android, and/or desktop): iOS
Scripting Runtime (Mono, and/or IL2CPP): IL2CPP

### Please describe the issue here:
(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)

1. Create 2019.3.8 project w/ URP template
2. Add ARFoundation (+ ARKit, ARCore) via Package Manager
3. Switch to iOS, Build and Run
4. _Everything builds fine_
5. Add Firebase Auth via Package Manager
6. Build and Run, build fails with an error:

```
Error running CocoaPods. Please ensure you have at least version 1.0.0.  You can install CocoaPods with the Ruby gem package manager:
 > sudo gem install -n /usr/local/bin cocoapods
 > pod setup

'pod --version' returned status: 1
output: 

/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/dependency.rb:313:in `to_specs': Could not find 'minitest' (~> 5.1) - did find: [minitest-4.7.4] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=/Users/jonah/.rvm/gems/ruby-1.9.2-p290:/Users/jonah/.rvm/gems/ruby-1.9.2-p290@global:/usr/local/Cellar/cocoapods/1.9.1/libexec', execute `gem env` for more information
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1449:in `block in activate_dependencies'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1452:in `block in activate_dependencies'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1452:in `block in activate_dependencies'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1438:in `activate_dependencies'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/specification.rb:1420:in `activate'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems.rb:304:in `block in activate_bin_path'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems.rb:303:in `synchronize'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems.rb:303:in `activate_bin_path'
	from /usr/local/Cellar/cocoapods/1.9.1/libexec/bin/pod:23:in `<main>'

UnityEngine.Debug:LogError(Object)
Google.Logger:Log(String, LogLevel) (at Z:/tmp/tmp.1ZjQmjdN1r/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/Logger.cs:136)
Google.IOSResolver:Log(String, Boolean, LogLevel) (at Z:/tmp/tmp.KKaSQttT3P/third_party/unity/unity_jar_resolver/source/IOSResolver/src/IOSResolver.cs:1013)
Google.IOSResolver:OnPostProcessInstallPods
```

**In the past, I've tried installing the Firebase packages via the ".unitypackage" files** files, but in those cases I've been hit with insurmountable other errors instead when building to Android. I just returned to that old project and I'm seeing the same Cocoapods error regardless. 

Perhaps it's relevant that I have Cocoapods installed via Homebrew because installing via gem raises errors. Apparently there's often this issue on OSX Catalina, [with the Homebrew fix discussed here](https://github.com/CocoaPods/CocoaPods/issues/8955#issuecomment-539267256). But regardless, everything builds fine to iOS until adding any Firebase package (apparently by either means).

### Please answer the following, if applicable:
Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?
To recreate the problem I need to switch the starter project to 2019.3.8. I'm then prompted immediately to upgrade the Asset Database to V2, and upon doing so there are lots of errors (e.g. Assets/Firebase/Sample/Auth/UIHandler.cs(20,18): error CS0234: The type or namespace name 'Extensions' does not exist in the namespace 'Firebase' (are you missing an assembly reference?)

To replicate, I'm assuming I'd need to be within 2019.3.8 and using ARFoundation + URP, which are necessary for my project.

### What's the issue repro rate? (eg 100%, 1/5 etc)
**100%**

**Comment by DellaBitta** [comment]
Hi @jonahgoldsaito,

the team had similar issues when upgrading to Catalina. To fix it we had to uninstall our older cocapods and reinstall them from scratch.  We took a sort of .. clean slate approach.

```
> sudo gem uninstall -aIx
> rm -rf ~/.gem
> sudo gem install cocoapods
```

Some of us also had to run chown on some directories to grant user rights.  If you run into a permissions problem in this process then you might have success changing the user ownership of the directories it complains about, such as

```
sudo chown -R $USER ~/Library/Caches/CocoaPods
```

I hope this helps, and let us know how it goes.

**Comment by jonahgoldsaito** [comment]
Thanks for the reply, @DellaBitta.

I'm continuing to uninstall and reinstall Cocoapods (dumped the Homebrew version I'd had to use initially, installed via gem, then uninstalled and reinstalled using Unity's iOS Resolved > Install CocoaPods).

The initial error is gone, though now each time I attempt an iOS build, Unity gives two interesting messages (after hanging for a while in the "Converting managed assemblies to C++" stage):

1. It gives an error that the "iOS framework addition failed due to CocoaPods installation failure: this will likely result in a non-functional Xcode project." _The full error message at bottom_
2. Followed by successful build message, which seems a little at odds with the first :)

An odd thing that jumps out at me in the full output is that it mentions:
`Firebase/Auth (= 6.21.0)`
But I'm using Firebase 6.14.1, and I'm assuming 6.21.0 hasn't been released yet unless I'm (very likely) misunderstanding the message.

I've also added export LANG=en_US.UTF-8 to my .profile and rebooted for good measure as it prompts, and it continues to give me that error. Interesting.

Lastly, does this combo of msgs mean that the build is otherwise OK and that I can add missing frameworks/etc manually to the Xcode project? Not ideal, but if necessary is it clear what those would be?

Thanks again :)

Full output:
```
iOS framework addition failed due to a CocoaPods installation failure. This will will likely result in an non-functional Xcode project.

After the failure, "pod repo update" was executed and succeeded. "pod install" was then attempted again, and still failed. This may be due to a broken CocoaPods installation. See: https://guides.cocoapods.org/using/troubleshooting.html for potential solutions.

pod install output:

Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "Firebase/Auth":
  In Podfile:
    Firebase/Auth (= 6.21.0)

None of your spec sources contain a spec satisfying the dependency: `Firebase/Auth (= 6.21.0)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.


    WARNING: CocoaPods requires your terminal to be using UTF-8 encoding.
    Consider adding the following to ~/.profile:

    export LANG=en_US.UTF-8
    [0m



pod repo update output:

Updating spec repo `cocoapods`

    WARNING: CocoaPods requires your terminal to be using UTF-8 encoding.
    Consider adding the following to ~/.profile:

    export LANG=en_US.UTF-8
    [0m

UnityEngine.Debug:LogError(Object)
Google.Logger:Log(String, LogLevel) (at Z:/tmp/tmp.1ZjQmjdN1r/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/Logger.cs:136)
Google.IOSResolver:Log(String, Boolean, LogLevel) (at Z:/tmp/tmp.KKaSQttT3P/third_party/unity/unity_jar_resolver/source/IOSResolver/src/IOSResolver.cs:1013)
Google.IOSResolver:OnPostProcessInstallPods(BuildTarget, String) (at Z:/tmp/tmp.KKaSQttT3P/third_party/unity/unity_jar_resolver/source/IOSResolver/src/IOSResolver.cs:2338)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr) (at /Users/builduser/buildslave/unity/build/Modules/IMGUI/GUIUtility.cs:187)

```

**Comment by DellaBitta** [comment]
Hi @jonahgoldsaito,

> But I'm using Firebase 6.14.1, and I'm assuming 6.21.0 hasn't been released yet unless I'm (very likely) misunderstanding the message.

Yep! Firebase for Unity 6.14.1 wraps the iOS and Android implementations of Firebase, which have their own version numbers.  In this case 6.14 of Unity Firebase depends upon 6.21 of Firebase for iOS.

> I've also added export LANG=en_US.UTF-8 to my .profile and rebooted for good measure as it prompts, and it continues to give me that error. Interesting.

It sounds like you ~/.profile isn't being sourced properly on login.  Instead you can try just adding the environment variable to the command line when you run your pod commands.  For instance:

`> LANG=en_US.UTF-8 pod repo update `

Hope this helps.

**Comment by jonahgoldsaito** [comment]
**I just upgraded to 6.15.0 under a new branch, and it's now building normally to iOS and Android.** Unfortunately the first build stalled out and the second spent 30m in "Converting managed assemblies to C++," but subsequent builds seem to be faster. So,  good signs! 

It doesn't help solve the issue for anyone locked into 6.14.1, but I'm in a position where I think I can upgrade safely. Perhaps knowing the fix is embedded in 6.15.0 helps you all. Finger crossed :) 🤞 

Unfortunately I'm now hitting the in-Editor 2nd Build crash [bug](https://github.com/firebase/quickstart-unity/issues/698). I'll subscribe to that issue. Good luck to the team! :)

**Comment by DellaBitta** [comment]
Hi @jonahgoldsaito,

I'm going to close this ticket since this particular issue has been resolved for you (though I'm sorry. you're hitting #698, too).  Please feel free to reopen this and mention me if you have any more information or questions.  Thanks.
